### PR TITLE
Added an example for infer_freq

### PR DESCRIPTION
**Pull request by Remit**
Added an example for infer_freq function. The initial index is built using a fixed number of periods to avoid encoding frequency in this index explicitly.

### CODE REVIEW

File: pandas/tseries/frequencies.py [hunk 1]
```diff
@@ -139,6 +139,12 @@ def infer_freq(index, warn: bool = True) -> Optional[str]:
         If the index is not datetime-like.
     ValueError
         If there are fewer than three values.
+
+    Examples
+    --------
+    >>> idx = pd.date_range(start = '2020/12/01', end = '2020/12/30', periods = 30)
```
Comment (mroeschke) [inline pandas/tseries/frequencies.py:145]: Could you eliminate the spaces between the `=` e.g `start = '2020/12/01'`

**Review by mroeschke — CHANGES_REQUESTED** [review]
Sorry for letting this PR slip through the cracks. Just one comment otherwise LGTM

**Comment by mroeschke** [comment]
Also this PR should target the master branch instead

**Comment by jreback** [comment]
let's close this in a couple of days if no response

File: pandas/tseries/frequencies.py (hunk 1 above)
Comment (Remit) [reply pandas/tseries/frequencies.py]: @mroeschke done. I'm kinda newb to submitting my changes in public repos so I hope that I managed to do it correctly :)

**Comment by mroeschke** [comment]
@Remit I changed the target branch to master, but it will require you to rebase with master

**Comment by Remit** [comment]
> @Remit I changed the target branch to master, but it will require you to rebase with master

@mroeschke  thanks! I did the rebase, but it seems like some checks fail for the current build. Does not look to be connected with my commit

**Comment by jreback** [comment]
thanks @Remit 

the failures on default are known

**Comment by Remit** [comment]
> thanks @Remit
> 
> the failures on default are known

thanks Jeff! Got it.

**Pull request merged**